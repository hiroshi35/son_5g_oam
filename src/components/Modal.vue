<template>
  <div class="modal fade" :id="id" tabindex="-1" role="dialog">
    <div class="modal-dialog" :class="{'modal-lg': large,'modal-sm': small}">
      <div class="modal-content" ref="modal">
        <div class="modal-header">
          <button type="button" class="close" @click="closeModal"><span>&times;</span></button>
          <slot name="title"></slot>
        </div>
        <div class="modal-body">
          <slot></slot>
        </div>
        <div class="modal-footer">
          <slot name="footer"></slot>
        </div>
      </div>
    </div>
  </div>
</template>
<script>
  export default {
    props: {
      id: {
        required: true
      },
      large: {
        default: false
      },
      small: {
        default: false
      },
      show: {
        default: false
      }
    },
    watch: {
      show (val) {
        if (val) {
          $(this.$el).modal('show')
          document.addEventListener('click', this.clickOutside, false);
        } else {
          $(this.$el).modal('hide')
          document.removeEventListener('click', this.clickOutside, false)
        }
      }
    },
    mounted () {
      if (this.show) {
        $(this.$el).modal('show')
        document.addEventListener('click', this.clickOutside, false);
      }
    },
    methods: {
      closeModal () {
        this.$emit('modalClosed');
      },
      clickOutside (e) {
        if (!this.$refs.modal.contains(e.target)) {
          this.closeModal();
        }
      }
    }
  }
</script>
