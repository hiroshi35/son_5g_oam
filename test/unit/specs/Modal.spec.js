import { shallow } from 'vue-test-utils'
import Modal from '@/components/Modal.vue'
import {modal} from 'jquery'

describe('Modal.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Modal,
      {
        slots: {
          default: 'Body',
          title: 'Title',
          footer: 'Footer'
        },
        propsData: {
          id: 'modal',
          show: true
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  test('modal is open', () => {
    expect(modal).toBeCalledWith('show')
  })
  test('event emitted when closing modal', () => {
    wrapper.find('.close').trigger('click')
    expect(wrapper.emitted().modalClosed).toBeTruthy()
  })
  test('$().modal("hide") should called when show = false', () => {
    wrapper.setProps({show: false})
    expect(modal).toBeCalledWith('hide')
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has correct props', () => {
    expect(wrapper.props().id).toBe('modal')
  })
})
