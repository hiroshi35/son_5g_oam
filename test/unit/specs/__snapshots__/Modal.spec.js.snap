// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal.vue has the expected html structure 1`] = `
<div
  class="modal fade"
  id="modal"
  role="dialog"
  tabindex="-1"
>
  <div
    class="modal-dialog"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <button
          class="close"
          type="button"
        >
          <span>
            Ã—
          </span>
        </button>
         Title
      </div>
       
      <div
        class="modal-body"
      >
        Body
      </div>
       
      <div
        class="modal-footer"
      >
        Footer
      </div>
    </div>
  </div>
</div>
`;
